extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -1359,35 +1360,43 @@ export class AmpStoryEmbeddedComponent {
 
   /**
    * Builds the focused state template.
-   * @param {!Document} doc
    * @return {!Element}
    * @private
    */
-  buildFocusedStateTemplate_(doc) {
-    const html = htmlFor(doc);
-    const tooltipOverlay = html`
+  renderFocusedStateElement_() {
+    const tooltipOverlay = (
       <section
-        class="i-amphtml-story-focused-state-layer
-            i-amphtml-story-system-reset i-amphtml-hidden"
+        class={objstr({
```

Comment:
Changed to `objstr` since the spaces in the class string were wasteful.

([`objstr` is completely free since we compile it away.](https://gist.github.com/alanorozco/4b7b9867d1b11c672939a8c3cb5d444e))

Reply:
I don't understand how the spaces are wasteful (but I also don't understand the Babel plugin in depth). How does this save space?

To me this looks like it will compile to `(true ? ' long-class-name' : '') + (true ? ' other-class-name' : '') + ...`
instead of `'long-class-name other-class-name'`, giving it more work to run on the browser and a larger bundle size.

Anything I'm missing?

Reply:
The classname is passed as:
```
i-amphtml-story-focused-state-layer
            i-amphtml-story-system-reset i-amphtml-hidden
```
So the newline + indentation spaces are preserved.

> To me this looks like it will compile to `(true ? ' long-class-name' : '') + (true ? ' other-class-name' : '') + ...`

The `obj-str` babel plugin evaluates these before inserting, so the expression is compressed when these are `true`:

`long-class-name other-classname`

Reply:
> Got it, that's unfortunate (and I assume it wouldn't be easy to solve from the compiler, because that would be the easiest for devx to avoid this in the future). 

You could potentially lint for this now since a `JSXAttribute[name.name=class]` has a single intent. 

> Does this happen only on Preact?

No, it's every string that's formatted with extra space when it doesn't need it. `htmlFor` was treated specially since we know that it always contains HTML, so we can minify it accordingly.

> Then it's ok to use `objstr`, or it would also work if you do 
> ```
> {'long-class-name' + 
>                 'other-class-name'}
> ```

Correct, that'd be fine to do. In this case the expression evaluation is done by Closure rather than the `objstr` plugin, but the result is the same.

> Just trying to understand the workings of this so we avoid having extra spaces in the future :)

Sure! I wouldn't worry about it too much though. I've only found two or three cases of this in the entire bundle and spaces are cheap (they're in the compression table).